### PR TITLE
Dividing up the various types and interfaces in p2p_messages.go

### PR DESCRIPTION
**Pull request by joeabbey**
**TL;DR:**
This is mostly superficial.. but the p2p_messages.go file was pretty big.  So I made smaller files.

**The DR:**
I was trying to use [Coverage Gutters](https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters) with this package and many of the files were shown as uncovered.  Upon inspecting further, I wondered if the various interface implementations were resulting in a coverage.xml that couldn't be applied.  

So I started dividing the files up into separate smaller scoped files.  This mostly worked but for whatever reason global scoped methods wouldn't show up either.  Then I discovered: 
image 

🤦‍♂ 

Apparently Visual Studio code has a built-in coverage for go coverage that one can simply toggle on.

Nevertheless, the work has been done to divide the p2p_messages.go and p2p_messages_test.go into smaller files.  

(I also added a Makefile, but there's nothing to really install b/c as I presume this is largely a module for other tooling.. just makes it easier to run `make` from a shell to run lint, vet, and test.)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+PROJECT := $(shell go list ./... | head -n 1)
+
+all: lint vet test
+
+.PHONY: help
+help: ## print this help
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```
[the remark this answers is not in the document]
Comment (gsgalloway) [reply Makefile]: Ah it's because no other make target is documented

File: constants.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
+package tezosprotocol
```
[the remark this answers is not in the document]
Comment (joeabbey) [reply constants.go]: Yeah, I'm find with relocating them towards having a tighter sense of locality.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (joeabbey) [reply Makefile]: yah.. I'm not super attached to it.. it's more helpful when there's non `.PHONY` targets :)

File: base58check_test.go [hunk 3]
```diff
@@ -4,7 +4,7 @@ import (
 	"encoding/hex"
 	"testing"
 
-	tezosprotocol "github.com/anchorageoss/tezosprotocol/v2"
+	"github.com/anchorageoss/tezosprotocol/v2"
```
[the remark this answers is not in the document]
Comment (joeabbey) [reply base58check_test.go]: Agreed, but I think there's some work to make v2 honored to:

https://blog.golang.org/v2-go-modules

**Comment by gsgalloway** [comment]
The codecov patch coverage threshold here is quite evidently not helpful. Will relax this constraint shortly
